apps/storybook/src/Table.stories.tsx

```diff
@@ -554,7 +564,12 @@ export const Expandable = () => {
         )}`,
       )(),
     [],
-  );
+  ) satisfies NonNullable<
+    TableProps<{
+      name: string;
+      description: string;
+    }>['onExpand']
+  >;
```

Comment:
can we make some of these less verbose by doing this?
```suggestion
  ) satisfies NonNullable<TableProps['onExpand']>;
```

if the user needs additional type checking, they can add the extra data types, but shouldn't be required to.

Reply:
i also noticed that `state` is `any` for some reason
<img width="287" alt="" src="https://github.com/iTwin/iTwinUI/assets/9084735/e2cea23d-c8bf-4b1e-a4e6-2f6cb6b316c4">

Reply:
https://github.com/iTwin/iTwinUI/pull/1514/commits/234f4f84535b9a7581400d3b4869fbc868304739: I removed all such `satisfies NonNullable<...>` that used `{ name: string; description; }`. But I didn't remove the ones that already use `TableStoryDataType` or something similar because they were not too verbose. That way we can show some variations while still not being verbose.

Reply:
> i also noticed that `state` is `any` for some reason

Looks like this is happening when `state` is assigned a type from `react-table-types.ts` itself.

<!-- ![image](https://github.com/iTwin/iTwinUI/assets/45748283/411de7a7-aec5-45c1-8ed1-b0e0bf5a1701) -->

![image](https://github.com/iTwin/iTwinUI/assets/45748283/5b2fdf4c-ef32-48f6-a9bb-b56dbd71fa5a)

Because when I change it to some primitive type like `string` or `number`, the type of state is inferred correctly.

<!-- ![image](https://github.com/iTwin/iTwinUI/assets/45748283/82a3f3ed-e554-4381-8df6-8ab681c07a38) -->

![image](https://github.com/iTwin/iTwinUI/assets/45748283/e968c76a-c5d1-4a8c-a813-dc12cbc29a0a)


<!-- ![image](https://github.com/iTwin/iTwinUI/assets/45748283/c8ba2dd0-d0db-4626-ae4e-826ab4afb4de) -->

I haven't been able to find a fix so far, but just wanted to give a quick update while I am still searching for the fix.

Reply:
I realized that this was happening because in `Table.tsx` we were importing `react-table`, instead of importing by path.

E.g. in `Table.d.ts` generated by tsc using `Table.tsx`
```diff
- import type { CellProps, TableOptions, Row, TableState } from 'react-table';
+ import type { CellProps, TableOptions, Row, TableState } from../../react-table/react-table.js
```

Importing from `react-table` from this file imports from the real `react-table` package instead of from the path mentioned in `paths` of `tsconfig`.

I did this commit (https://github.com/iTwin/iTwinUI/pull/1514/commits/7b4c2c8fd071910ea366c7196d6c512e6333a7f8) to fix the above problem. I noticed quite a few type errors in `Table.stories.tsx`. To avoid convoluting this PR, I reverted that commit and will fix the above issue in another PR.

Reply:
fine by me